nav2_behavior_tree/plugins/control/sequence_with_blackboard_memory.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2019 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "nav2_behavior_tree/plugins/control/sequence_with_blackboard_memory.hpp"
+#include "behaviortree_cpp/action_node.h"
+#include "behaviortree_cpp/bt_factory.h"
+
+namespace nav2_behavior_tree
+{
+
+SequenceWithBlackboardMemoryNode::SequenceWithBlackboardMemoryNode(
+  const std::string & name,
+  const BT::NodeConfiguration & conf)
+: BT::ControlNode::ControlNode(name, conf)
+{
+  setRegistrationID("SequenceWithBlackboardMemory");
+}
+
+void SequenceWithBlackboardMemoryNode::halt()
+{
+  ControlNode::halt();
+}
+
+BT::NodeStatus SequenceWithBlackboardMemoryNode::tick()
+{
+  const int children_count = children_nodes_.size();
+
+  int current_child_idx;
+  getInput("current_child_idx", current_child_idx);
+
+  setStatus(BT::NodeStatus::RUNNING);
+
+  while (current_child_idx < children_count) {
+    TreeNode * current_child_node = children_nodes_[current_child_idx];
+    const BT::NodeStatus child_status = current_child_node->executeTick();
+
+    switch (child_status) {
+      case BT::NodeStatus::RUNNING: {
+          return child_status;
+        }
+      case BT::NodeStatus::FAILURE: {
+          // Reset on failure
+          resetChildren();
+          current_child_idx = 0;
+          setOutput("current_child_idx", 0);
+          return child_status;
+        }
+      case BT::NodeStatus::SUCCESS: {
+          current_child_idx++;
+          setOutput("current_child_idx", current_child_idx);
+        }
+        break;
+
+      case BT::NodeStatus::SKIPPED: {
```

Comment:
I think we mostly treat skipped as a failure in other control nodes: https://github.com/ros-navigation/navigation2/blob/d438a6e186e449a2977a9e6188cdbd0c99a1d307/nav2_behavior_tree/plugins/control/recovery_node.cpp#L83-L92